lib/mapObjects/CGTownInstance.cpp

```diff
@@ -1206,6 +1206,22 @@ void CGTownInstance::serializeJsonOptions(JsonSerializeFormat & handler)
 	{
 		handler.serializeIdArray( "possibleSpells", possibleSpells);
 		handler.serializeIdArray( "obligatorySpells", obligatorySpells);
+
+		if (handler.saving) 
+		{
+			auto eventsHandler = handler.enterArray("events");
+			std::vector<CCastleEvent> temp(events.begin(), events.end());
+			eventsHandler.serializeStruct(temp);
+		}
+		else
+		{
+			auto eventsHandler = handler.enterArray("events");
+			std::vector<CCastleEvent> temp;
+			eventsHandler.serializeStruct(temp);
+			events.clear();
+			events.insert(events.begin(), temp.begin(), temp.end());
+		}
```

Comment:
Completely untested, but will something like this work?

```suggestion
	{
		auto eventsHandler = handler.enterArray("customSize");
		eventsHandler.syncSize(events, JsonNode::JsonType::DATA_VECTOR);
		eventsHandler.serializeStruct(events);
	}
```

Reply:
```
 no instance of overloaded function matches the argument list
argument types are: (std::list<CCastleEvent, std::allocator<CCastleEvent>>) 
object type is: JsonArraySerializer
 ```
 
 events are list, and serializeStruct expect vector

Reply:
Personally I would change list to vector.
Performance hit likely would be negligible (or even would have an improvement)